testcases/Build_bisector.py

```diff
@@ -0,0 +1,117 @@
+        self.repo = self.conf.args.git_repo
+        self.repo_reference = self.conf.args.git_repo_reference
+        self.branch = self.conf.args.git_branch
+        self.home = self.conf.args.git_home
+        self.config_path = self.conf.args.git_repoconfigpath
+        self.config = self.conf.args.git_repoconfig
+        self.good_commit = self.conf.args.good_commit
+        self.bad_commit = self.conf.args.bad_commit
+        self.bisect_script = self.conf.args.bisect_script
+        self.bisect_category = self.conf.args.bisect_category
+        self.append_kernel_cmdline = self.conf.args.append_kernel_cmdline
+        self.linux_path = os.path.join(self.home, "linux")
+        if self.config_path:
+            self.config = "olddefconfig"
+        if not self.repo:
+            self.fail("Provide git repo of kernel to install")
+        if not (self.conf.args.host_ip and self.conf.args.host_user and self.conf.args.host_password):
+            self.fail(
+                "Provide host ip user details refer, --host-{ip,user,password}")
+    def get_email(self, commit_id):
+        try :
+            self.connection.run_command("git config --global color.ui true")
+            result = self.connection.run_command("git show --format=%ce {} | sed -r 's/\x1B\[[0-9:]*[JKsu]//g'".format(commit_id))
+            print("result",result,"result")
+            email = result[0]
+            print("email: ",email)
+
+            return email
+        except subprocess.CalledProcessError as e:
+            print("ERROR",e)
+            return None    
+    def runTest(self):
+        self.connection.run_command(" if [ '$(pwd)' != {} ]; then cd {} || exit 1 ; fi ".format(self.linux_path,self.linux_path))
+        shallow = self.connection.run_command("git rev-parse --is-shallow-repository")
+        print(type(shallow[-1]))
+        if shallow[-1] == True or shallow[-1] == 'true':
+            self.connection.run_command("git fetch --unshallow",timeout=3000)
+        makefile_path = os.path.join(self.conf.basedir, "make.sh")
+        self.cv_HOST.copy_test_file_to_host(makefile_path, dstdir=self.linux_path)
+        self.connection.run_command("git bisect start")
+        self.connection.run_command("git bisect good {} ".format(self.good_commit))
+        self.connection.run_command(" git bisect bad ")
+        self.connection.run_command("chmod +x ./make.sh ")
+        commit =  self.connection.run_command(" git bisect run ./make.sh")
+        log.info(commit)
+        badCommit = [word for word in commit if word.endswith("is the first bad commit")]
+        badCommit= badCommit[0].split()[0]
+        log.info("BISECT")
+        log.info(badCommit)
+        email = self.get_email(badCommit)
+        subject = f"Author of Commit {badCommit}"
+        body = f"The above commit is causing build errors"
+        self.connection.run_command("git bisect reset")
+        log.info("email  ")
+        log.info(email)  
```

Comment:
can have a single log.info to capture logs

Reply:
done